### DISCUSSION TITLE
specifying an ID will cause problems

### DISCUSSION THREAD

**Discussion by freemant2000** [elimintz/justpy#577] [General]
Hi,

If the html ID is set and event handling is enabled for that component, then there will be problem as different component instances on different page instances have the same ID, which is used in a global dict:
```
    def on(self, event_type, func, *, debounce=None, throttle=None, immediate=False, meadows=False):
        if meadows or event_type in self.allowed_events:
            cls = JustpyBaseComponent
            if not self.id:
                self.id = cls.next_id
                cls.next_id += 1
            cls.instances[self.id] = self <---- use the element id in a global dict
```
perhaps we should use both the page id and element id for this?

**Comment by sandeep-gh** [comment]
The ID generation and management needs to be revisited. I think we should refactor out an ID generator and allow for plugable ID generators. This will also fix the problem I am having with WTFor data validators. 
@WolfgangFahl : Does this sound ok?
